[DM-3800][OPCUA] Fix local db memory issue (#2365)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-10-18-524-0-fix-local-database-memory-issue.md

---
date:

> title: Fixed local database memory issue for OPC UA gateway

Reviewer: ```suggestion
title: New cleanup process during the gateway startup to reduce the size of the database file
```

Author: @BeateRixen Thanks for the suggestion. After reading the title I realized that I missed a crucial part of the description...that it is a one-time job. So I updated both the description and title.

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3800
version: 10.18.524.0
---

> The OPC UA gateway creates a local database namely 'cumulocity-opcua-gateway.db' to store essential data, including a list of executed operation IDs.

Reviewer: ```suggestion
The OPC UA gateway creates a local database `cumulocity-opcua-gateway.db` to store essential data, including a list of executed operation IDs.
```

---
date:
title: Optimization of storage management for OPC UA gateway
product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3800
version: 10.18.524.0
---
The OPC UA gateway creates a local database `cumulocity-opcua-gateway.db` to store essential data, including a list of executed operation IDs.

> As this list expands, the database file consumes more disk space. This issue has been resolved with the latest update by introducing an in-memory cache to store executed operation IDs with each entry set auto-expire after 24 hours. 

Reviewer: ```suggestion
As this list expands, the database file consumes more disk space. This issue has been resolved by introducing an in-memory cache to store executed operation IDs with each entry set to auto-expire after 24 hours. 
```
